pkg/linter/linter.go

```diff
@@ -152,10 +152,38 @@ func New(cxt *portercontext.Context, mixins pkgmgmt.PackageManager) *Linter {
 	}
 }
 
+
 func (l *Linter) Lint(ctx context.Context, m *manifest.Manifest) (Results, error) {
 	// TODO: perform any porter level linting
 	// e.g. metadata, credentials, properties, outputs, dependencies, etc
+	reservedPrefixes := []string{"porter-", "porter_"}
+	params := m.Parameters
+
+	var results Results
 
+	for _, param := range params {
+		paramName := strings.ToLower(param.Name)
+		for _, reservedPrefix := range reservedPrefixes {
+			if strings.HasPrefix(paramName, reservedPrefix) {
+
+				res := Result {
+					Level: LevelWarning,
+					Location: Location {
+						Action: "",
+						Mixin: "",
+						StepNumber: 0,
+						StepDescription: "",
+				
+					},
+					Code: "exec-100",
+					Title: "Reserved name warning",
+					Message: param.Name + " has a reserved prefix",
+					URL: "",
```

Comment:
I've created another issue that builds off of what you have done here. #2486 Feel free to take that issue once you are finished with this pull request if you are interested!

Reply:
Nevermind, someone else asked to work on that issue. But once they are done we will need to add your new lint error to the page that they create.